lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -424,17 +426,19 @@ def create_test_xcconfig_files(test_native_targets, test_resource_bundle_targets
 
           # Creates a script that copies the resources to the bundle of the test target.
           #
-          # @param [Symbol] test_type
-          #        The test type to create the script for.
+          # @param [Specification] test_spec
+          #        The test spec to create the copy resources script for.
           #
           # @return [void]
           #
-          def create_test_target_copy_resources_script(test_type)
-            path = target.copy_resources_script_path_for_test_type(test_type)
-            pod_targets = target.all_dependent_targets
+          def create_test_target_copy_resources_script(test_spec)
+            path = target.copy_resources_script_path_for_test_spec(test_spec)
+            pod_targets = target.dependent_targets_for_test_spec(test_spec)
             resource_paths_by_config = target.user_build_configurations.keys.each_with_object({}) do |config, resources_by_config|
               resources_by_config[config] = pod_targets.flat_map do |pod_target|
-                spec_paths_to_include = pod_target == target ? pod_target.specs.map(&:name) : pod_target.non_test_specs.map(&:name)
+                spec_paths_to_include = pod_target == target ? pod_target.specs.select do |spec|
```

Comment:
this logic is a bit hard to follow, should we extract a method out to simplify it?

Reply:
it's all been used in many places.

Reply:
probably a good call